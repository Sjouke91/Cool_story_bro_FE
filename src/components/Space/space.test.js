import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Space from "../Space";

const props = {
  id: 9,
  title: "this is test",
  description: "testing all",
  showLink: true,
  color: "#FFFFFF",
  backgroundColor: "#FFFFFF",
};

// describe("<Space >", () => {
//   test("should render without props", () => {
//     const component = renderer.create(<Space />); // "render" the component
//     const tree = component.toJSON(); // Convert it to JSON
//     expect(tree).toMatchSnapshot(); // Compare to stored snapshot
//   });
// });

// describe("<Space >", () => {
//   test("should render with props", () => {
//     const component = renderer.create(<Space {...props} />); // "render" the component
//     const tree = component.toJSON(); // Convert it to JSON
//     expect(tree).toMatchSnapshot(); // Compare to stored snapshot
//   });
// });

describe("<Space >", () => {
  test("should render with props and link to true", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Space {...props} />
      </MemoryRouter>
    ); // "render" the component
    const tree = component.toJSON(); // Convert it to JSON
    expect(tree).toMatchSnapshot(); // Compare to stored snapshot
  });
});
