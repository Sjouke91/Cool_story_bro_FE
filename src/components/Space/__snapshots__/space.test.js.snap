// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space > should render with props and link to true 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "color": "#FFFFFF",
    }
  }
>
  <h1>
    this is test
  </h1>
  <p>
    testing all
  </p>
  <a
    href="/spaces/9"
    onClick={[Function]}
  >
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Visit space
    </button>
  </a>
</div>
`;
